nautobot/core/cli.py

```diff
@@ -132,9 +132,12 @@ def _configure_settings(config):
     if settings.METRICS_ENABLED and "postgres" in settings.DATABASES["default"]["ENGINE"]:
         settings.DATABASES["default"]["ENGINE"] = "django_prometheus.db.backends.postgresql"
 
-    # Create fake db for job logs. This uses the default db, but allows us to save logs within
-    # transaction.atomic().
-    settings.DATABASES["job_logs"] = settings.DATABASES["default"]
+    # Create secondary db connection for job logging. This still writes to the default db, but because it's a separate
+    # connection, it allows allows us to "escape" from transaction.atomic() and ensure that job log entries are saved
+    # to the database even when the rest of the job transaction is rolled back.
+    settings.DATABASES["job_logs"] = settings.DATABASES["default"].copy()
+    # When running unit tests, treat it as a mirror of the default test DB, not a separate test DB of its own
+    settings.DATABASES["job_logs"]["TEST"] = {"MIRROR": "default"}
```

Comment:
It is magic - unless we want to push this to the user-facing configs in `nautobot_config.py`, it has to be done here rather than in `settings.py` because it needs to be done *after* `DATABASES` is defined in `nautobot_config.py`. 

We could move it to `nautobot_config.py` for 1.3 if we really want to, but 99.9% of the time `settings.DATABASES["job_logs"]` *should* just be a copy of `settings.DATABASES["default"]` plus the `TEST` settings, so requiring the user to set it is just a bit of an extra burden on the user. Let me know what you'd prefer.

Reply:
One possibility could be to make this fixup conditional on the user not having already defined it, i.e.

```python
if "job_logs" not in settings.DATABASES:
    settings.DATABASES["job_logs"] = settings.DATABASES["default"].copy()
if "TEST" not in settings.DATABASES["job_logs"]:
    settings.DATABASES["job_logs"]["TEST"] = {"MIRROR": "default"}
```

I don't want to only add the TEST settings into `nautobot/core/tests/nautobot_config.py`, because any other implementers of Job tests (i.e. plugins) will also need to have this setting for their tests to work properly as well. It's harmless to have set in non-testing environments.

Reply:
We discussed this further and concluded that it's OK in its current form at this time.